consensus/consensus_leader.go

```diff
@@ -279,142 +225,95 @@ func (consensus *Consensus) processResponseMessage(message consensus_proto.Messa
 	messageBytes, err := protobuf.Marshal(&message)
 	if err != nil {
-		utils.GetLogInstance().Warn("Failed to marshal the announce message", "error", err)
-	}
-	if schnorr.Verify(crypto.Ed25519Curve, value.PubKey, messageBytes, signature) != nil {
-		utils.GetLogInstance().Warn("Received message with invalid signature", "validatorKey", consensus.leader.PubKey, "consensus", consensus)
-		return
+		utils.GetLogInstance().Warn("Failed to marshal the commit message", "error", err)
 	}
+	_ = messageBytes
+	_ = signature
+	// TODO: verify message signature
+	//if schnorr.Verify(crypto.Ed25519Curve, value.PubKey, messageBytes, signature) != nil {
+	//	consensus.Log.Warn("Received message with invalid signature", "validatorKey", consensus.leader.PubKey, "consensus", consensus)
+	//	return
+	//}
 
-	commitments := consensus.commitments // targetState == CollectiveSigDone
-	responses := consensus.responses
-	bitmap := consensus.bitmap
-	if targetState == Finished {
-		commitments = consensus.finalCommitments
-		responses = consensus.finalResponses
-		bitmap = consensus.finalBitmap
-	}
+	commitSigs := consensus.commitSigs
+	commitBitmap := consensus.commitBitmap
 
 	// proceed only when the message is not received before
-	_, ok = (*responses)[validatorID]
+	_, ok = (*commitSigs)[validatorID]
 	shouldProcess = shouldProcess && !ok
 
-	if len((*responses)) >= ((len(consensus.PublicKeys)*2)/3 + 1) {
+	if len((*commitSigs)) >= ((len(consensus.PublicKeys)*2)/3 + 1) {
 		shouldProcess = false
 	}
 
 	if shouldProcess {
-		// verify the response matches the received commit
-		responseScalar := crypto.Ed25519Curve.Scalar()
-		responseScalar.UnmarshalBinary(response)
-		err := consensus.verifyResponse(commitments, responseScalar, validatorID)
+		var sign bls.Sign
+		err := sign.Deserialize(commitSig)
 		if err != nil {
-			utils.GetLogInstance().Warn("leader failed to verify the response", "error", err, "VID", strconv.Itoa(int(validatorID)))
-			shouldProcess = false
-		} else {
-			(*responses)[validatorID] = responseScalar
-			utils.GetLogInstance().Debug("Received new response message", "num", len(*responses), "validatorID", strconv.Itoa(int(validatorID)))
-			// Set the bitmap indicate this validate signed.
-			bitmap.SetKey(value.PubKey, true)
+			utils.GetLogInstance().Debug("Failed to deserialize bls signature", "validatorID", validatorID)
 		}
+		// TODO: check bls signature
+		(*commitSigs)[validatorID] = &sign
+		utils.GetLogInstance().Debug("Received new commit message", "numReceivedSoFar", len(*commitSigs), "validatorID", strconv.Itoa(int(validatorID)))
+		// Set the bitmap indicate this validate signed.
+		commitBitmap.SetKey(value.PubKey, true)
 	}
 
 	if !shouldProcess {
-		utils.GetLogInstance().Debug("Received new response message", "validatorID", strconv.Itoa(int(validatorID)))
+		utils.GetLogInstance().Debug("Received additional new commit message", "validatorID", strconv.Itoa(int(validatorID)))
 		return
 	}
 
 	threshold := 2
```

Comment:
we don't need threshold = 1 anymore? If that's case, we can just remove this variable

Reply:
I will correct this in a following PR.